src/plugins/vis_builder/public/application/components/right_nav.tsx

```diff
@@ -18,28 +18,48 @@ import './side_nav.scss';
 import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
 import { VisBuilderServices } from '../../types';
 import {
+  ActiveVisPayload,
   setActiveVisualization,
   useTypedDispatch,
   useTypedSelector,
 } from '../utils/state_management';
-import { usePersistedAggParams } from '../utils/use/use_persisted_agg_params';
+import { getPersistedAggParams } from '../utils/get_persisted_agg_params';
 
 export const RightNav = () => {
-  const [newVisType, setNewVisType] = useState<string>();
+  const { ui, name: activeVisName } = useVisualizationType();
+  const [confirmAggs, setConfirmAggs] = useState<ActiveVisPayload | undefined>();
   const {
     services: { types },
   } = useOpenSearchDashboards<VisBuilderServices>();
-  const { ui, name: activeVisName } = useVisualizationType();
   const dispatch = useTypedDispatch();
   const StyleSection = ui.containerConfig.style.render;
 
   const { activeVisualization } = useTypedSelector((state) => state.visualization);
-  const aggConfigParams = activeVisualization?.aggConfigParams ?? [];
-  const persistedAggParams = usePersistedAggParams(
-    types,
-    aggConfigParams,
-    activeVisName,
-    newVisType
+  const aggConfigParams = useMemo(() => activeVisualization?.aggConfigParams ?? [], [
+    activeVisualization,
+  ]);
+
+  const handleVisTypeChange = useCallback(
+    (newVisName) => {
+      const currentVisSchemas = types.get(activeVisName)?.ui.containerConfig.data.schemas.all ?? [];
+      const newVisSchemas = types.get(newVisName)?.ui.containerConfig.data.schemas.all ?? [];
+      const persistedAggParams = getPersistedAggParams(
+        aggConfigParams,
+        currentVisSchemas,
+        newVisSchemas
+      );
+
+      const newVis = {
+        name: newVisName,
+        aggConfigParams: persistedAggParams,
+        style: types.get(newVisName)?.ui.containerConfig.style.defaults,
+      };
+
+      if (persistedAggParams.length !== aggConfigParams.length) return setConfirmAggs(newVis);
```

Comment:
To confirm my understanding: `persistedAggParams.length` will always be <= to `aggConfigParams.length`, and never greater? If that's the case, I think `<` is better than `!==`, particularly given the warning message about fields being lost.

Reply:
yes the length will never be greater, but is your concern with the check readability? since functionally both options are the same

Reply:
Yeah, I think the `<` should be explicit rather than implied.